File: jade/src/main/java/com/blockstream/jade/HttpRequestHandler.java
package com.blockstream.jade;

import com.fasterxml.jackson.databind.JsonNode;

import java.io.IOException;
import java.net.URL;
import java.util.List;

import kotlinx.serialization.json.JsonElement;

// HttpRequestHandler is used for network calls during pinserver handshake
// useful on TOR enabled sessions
public interface HttpRequestHandler{

    void prepareHttpRequest();
    JsonElement httpRequest(final JsonElement details) throws IOException;

    JsonElement httpRequest(final String method,
                                final List<URL> urls,
                                final String data,
                                final String accept,
                                final List<String> certs) throws IOException;
}

File: jade/src/main/java/com/blockstream/jade/HttpRequestHandler.kt
package com.blockstream.jade

import kotlinx.serialization.json.JsonElement
import java.io.IOException
import java.net.URL

// HttpRequestHandler is used for network calls during pinserver handshake
// useful on TOR enabled sessions
interface HttpRequestHandler {
    fun prepareHttpRequest()

    @Throws(IOException::class)
    fun httpRequest(details: JsonElement): JsonElement

    @Throws(IOException::class)
    fun httpRequest(
        method: String,
        urls: List<URL>?,
        data: String?,
        accept: String?,
        certs: List<String>?
    ): JsonElement
}